docs/auth.md

```diff
@@ -0,0 +1,149 @@
+
+kcp implements the same RBAC-based authorization scheme as Kubernetes.
+Other authorization schemes (i.e. ABAC) are currently not supported.
+Generally, the same role and role binding priniciples apply the same way as in Kubernetes.
+
+However RBAC rules are always declared within the context of a concrete workspace.
+Rules created in a workspace apply to requests against this workspace independently from the workspace type.
+This is what resemembles Kubernetes authorization model.
+In addition, rules can play a role in authorization to requests to other workspaces. The details will be described below.
+This is true for all non-system workspace types, including root, organizational and enduser.
+The exceptions are system namespaces. These can only be accessed by shard-local admins.
+
+# Authorizers
+
+## Workspace content authorizer
+
+As described above, RBAC rules apply for resources within workspaces.
+They cannot express the basic authorization primitive whether a subject is generally
+authorized to access a workspace.
+
+The lack of any bindings is not sufficient to express general access to a workspace
+as every authenticated user in Kubernetes is at least bound to `system:authenticated`
+with bindings against at least some cluster roles, like `system:basic-user`, `system:discovery`,
+and `system:public-info-viewer`.
+
+Expressing basic access to a workspace via bindings against the `ClusterWorkspace` resource type
+is not sufficient either because the permission to mutate the `ClusterWorkspace` resource
+(i.e. changing its metadata) is orthogonal to accessing its contained resources.
+
+Hence, an additional authorization layer is necessary. To express authorization patterns,
+kcp recognizes role bindings defined against the `clusterworkspaces/content` resource under the `tenancy.kcp.dev` api group.
+This subresource is not backed by any CRD but is known to the workspace content authorizer.
+
+Since `ClusterWorkspaces` are only defined in organizational workspaces,
+`clusterworkspaces/content` role bindings are expected to be contained in organizational workspaces as well.
+Concrete permissions can be declared using verbs.
+
+The workspace content authorizer ensures that permitted verbs are mapped to groups for a given subject.
+The assigned group is by default bound against standard Kubernetes cluster roles
+such that a simple binding against the `clusterworkspaces/content` resource
+implicitly grants permissions for contained resources within the workspace.
+The group assignments can then be used to refine permissions within workspaces with custom role bindings.
+
+Currently the following permissions (verbs) can be declared:
+
+| Verb     | Group                         | Bootstrap cluster rolebinding |
+| -------- | ----------------------------- | ----------------------------- |
+| `admin`  | `system:kcp:workspace:admin`  | `admin`                       |
+| `edit`   | `system:kcp:workspace:edit`   | `cluster-admin`               |
```

Comment:
where is the difference between admin and cluster-admin? Why does edit not map to some edit role?